### PR TITLE
Add support to DL server

### PR DESCRIPTION
**Pull request by yuvirani**
### Description
Add support to DL server

### Check List
- [ ] New functionality includes testing.
  - [ ]  All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [ ] New endpoints supported are updated in the endpoints-support.md file.
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/server_hardware/update_resource.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
+provider "oneview" {
+  ov_username   = var.username
+  ov_password   = var.password
+  ov_endpoint   = var.endpoint
+  ov_sslverify  = var.ssl_enabled
+  ov_apiversion = 3000
+  ov_ifmatch    = "*"
+}
+
```
Comment (AsisBagga) [inline examples/server_hardware/update_resource.tf:9]: does it allows to update all these attributes under one resource?

File: oneview/resource_server_hardware.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+	config := meta.(*Config)
+
+	servHard, err := config.ovClient.GetServerHardwareByName(d.Get("hostname").(string))
+	if err != nil || servHard.URI.IsNil() {
+		d.SetId("")
+		return nil
+	}
+	d.SetId(d.Id())
+	d.Set("configuration_state", d.Get("configuration_state").(string))
+	d.Set("host_name", servHard.Hostname)
```
Comment (AvijitOS) [inline oneview/resource_server_hardware.go:194]: do we have this in get call?

File: oneview/resource_server_hardware.go [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+	d.Set("server_hardware_type_uri", servHard.ServerHardwareTypeURI.String())
+	d.Set("server_profile_uri", servHard.ServerProfileURI.String())
+	d.Set("uuid", servHard.UUID.String())
+	d.Set("virtual_serial_number", servHard.VirtualSerialNumber.String())
+	d.Set("virtual_uuid", servHard.VirtualUUID)
+	d.Set("mp_ip_address", servHard.MpIpAddress)
+	d.Set("mp_firmware_version", servHard.MpFirwareVersion)
+	d.Set("mp_dns_name", servHard.MpDnsName)
+	d.Set("uid_state", servHard.UidState)
+	d.Set("user_name", servHard.Username)
```
Comment (AvijitOS) [inline oneview/resource_server_hardware.go:215]: not sure if it exists inside get call

File: examples/server_hardware/update_resource.tf (hunk 1 above)
Comment (yuvirani) [reply examples/server_hardware/update_resource.tf:9]: Updated the code to handle this.

File: oneview/resource_server_hardware.go [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+	config := meta.(*Config)
+
+	servHard, err := config.ovClient.GetServerHardwareByName(d.Get("hostname").(string))
+	if err != nil || servHard.URI.IsNil() {
+		d.SetId("")
+		return nil
+	}
+	d.SetId(d.Id())
+	d.Set("configuration_state", d.Get("configuration_state").(string))
+	d.Set("host_name", d.Get("configuration_state").(string))
```
Comment (AvijitOS) [inline oneview/resource_server_hardware.go:194]: this is should be d.get("host_name")

Comment (yuvirani) [reply oneview/resource_server_hardware.go]: Changed

**Review by AvijitOS — APPROVED** [review]

**Review by AsisBagga — APPROVED** [review]
LGTM

**Pull request merged**